phoenix-core/src/main/java/org/apache/phoenix/mapreduce/FormatToBytesWritableMapper.java

```diff
@@ -411,8 +421,46 @@ public void errorOnRecord(T record, Throwable throwable) {
             ImportPreUpsertKeyValueProcessor {
 
         @Override
-        public List<Cell> preUpsert(byte[] rowKey, List<Cell> keyValues) {
+        public List<Cell> preUpsert(byte[] tableName, List<Cell> keyValues) {
             return keyValues;
         }
     }
+
+    /**
+     * Updates the Empty cell value to VERIFIED for global index table rows
+     */
+    private static class IndexStatusUpdater {
+
+        private byte[] emptyKeyValueCF;
+        private int emptyKeyValueCFLength;
+        private byte[] emptyKeyValueQualifier;
+        private int emptyKeyValueQualifierLength;
+
+        public IndexStatusUpdater(byte[] emptyKeyValueCF, byte[] emptyKeyValueQualifier) {
+            this.emptyKeyValueCF = emptyKeyValueCF;
+            this.emptyKeyValueQualifier = emptyKeyValueQualifier;
+            emptyKeyValueCFLength = emptyKeyValueCF.length;
+            emptyKeyValueQualifierLength = emptyKeyValueQualifier.length;
+        }
+
+        /**
+         * Update the Empty cell values to VERIFIED in the passed keyValues list
+         * 
+         * @param keyValues will be modified
+         */
+        public void setVerfied(List<Cell> keyValues) {
+            for(int i=0; i < keyValues.size() ; i++) {
+                Cell kv = keyValues.get(i);
```

Comment:
can be replaced by
```
for (Cell kv : keyValues) {
```

Reply:
I purposefully chose this construct, as we're processing a lot a objects, and this is reportedly faster and easier on memory / GC.